src/runtime/xss_api.js

```diff
@@ -142,6 +142,41 @@ function sanitizeURL(url) {
   return '';
 }
 
+/**
+ * Sanitizes the specified attribute in the given array if present.
+ *
+ * @param {string} attribute the attribute to sanitize
+ * @param {*} attribs the attributes array to sanitize from
+ * @returns {object} the sanitized attribute and it's index in the array, or an empty object
+ */
+function sanitizeURLOnAttr(attribute, attribs) {
+  const index = attribs.indexOf(attribute);
+  if (index > -1) {
+    return { index, sanitizedUrl: sanitizeURL(attribs[index + 1]) || null };
+  }
+  return {};
+}
+
+/**
+ * A sanitization policy that validates src/href attributes against the URI scheme.
+ *
+ * @param {string} tagName The name of the tag currently parsed
+ * @param {string[]} attribs An array of attribute names and values
+ * @returns {object} the resulting sanitized attributes
+ */
+function sanitizeURLPolicy(tagName, attribs) {
+  const initial = [].concat(attribs);
```

Comment:
calling `makeTagPolicy` below actually modifies the input array and removes offending attributes, so we need to create an initial copy to save the links to be escaped below